third-party/jemalloc/Makefile

```diff
@@ -12,10 +12,19 @@ ifneq (, $(filter cray-x%,$(CHPL_MAKE_TARGET_PLATFORM)))
 CHPL_JEMALLOC_CFG_OPTIONS += --host=x86_64-cle-linux-gnu
 endif
 
-# Can't build under cce with `-hgnu` or anything above `-O1`. Additionally,
-# CCE doesn't support the dependency flags that jemalloc expects (-MM -MT)
+# Can't build under CCE with gcc extensions enabled or with high levels of
+# interprocedural analysis on. The build fails with -hipa3 (the default) or
+# higher so we limit to a max of -hipa2. Additionally, CCE doesn't have the
+# dependency flags that jemalloc expects (-MM -MT)
 ifeq ($(CHPL_MAKE_TARGET_COMPILER),cray-prgenv-cray)
-CFLAGS += -hnognu -O1
+CFLAGS += -hnognu
+
+ifeq ($(OPTIMIZE), 1)
+CFLAGS += -hipa2
+else
```

Comment:
I was torn on this. CCE's default is `-O2` which implies `-hipa3`. However, we currently [throw `-O0` for non optimized builds] (https://github.com/chapel-lang/chapel/blob/master/make/compiler/Makefile.cray-prgenv-cray#L33-L35). My tension was that we've tried and considered going back to `-O1` or `-hdevelop` for non optimized builds, so I didn't want that to unexpectedly bite us here.